web_programming/download_images_from_4chan.py

```diff
@@ -0,0 +1,108 @@
+import os
+import urllib
+
+import bs4 as bs
+import requests
+
+
+def get_links(url: str) -> list[str]:
+    """
+    Searches 4chan thread for all the available images
+
+    Args:
+        url: URL of 4chan thread
+    
+    Returns:
+        list containing all image URLs
+    """
+
+    print("Retrieving UR:s...")
+
+    page = requests.get(url)
+
+    soup = bs.BeautifulSoup(page.content, "html.parser")
+
+    results = soup.find_all(class_="fileText")
+
+    links = []
+
+    for i in results:
+        k = i.find("a")
+        links.append("https://" + k["href"][2:])
+
+    return links
+
+
+def save_links(links: list[str]) -> None:
+    """
+    Saves all image URLs to a file if user requests for it.
+
+    Args:
+        links: list containing all image URLs
+    
+    Returns:
+        None
+    """
+
+    print("Saving links...")
+
+    txt = ""
+    for i in links:
```

Comment:
```suggestion
    for link in links:
```
Use `i` for integers.